woocommerce/woocommerce
# Add new file dropdown menu

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #40493



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure feature `New product editor` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
2. Make sure feature `product-virtual-downloadable` is enabled under `Features` tab from `/wp-admin/tools.php?page=woocommerce-admin-test-helper` (WooCommerce Beta Tester plugin) is required
3. Go to `/wp-admin/admin.php?page=wc-admin&path=/add-product`
4. Under `General` tab and at the bottom of the page as part of the `Downloads` section a new `Add new` menu should be shown 
<img width="667" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/7f43d723-e96a-4408-9ad5-f90b7a1106b8">

5. Clicking the `Upload` menu item should let you upload files from the local computer
6. Clicking `Media Library` menu item should let you select/upload files from the media library WP modal
7. Clicking `Insert from URL` menu item should let you set an URL 
<img width="661" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/73f9b223-7d09-4079-a371-a8423d9b73f0">

8. Add as many files as you want combining points 5, 6 and 7 and then save the product
9. If you hard refresh (F5 or Cmd + R) the page all the uploaded files should remains listed in the `Downloads` table 
<img width="681" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/9c00eb4a-6a9a-4e26-963a-4014562326e7">




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
